.github/workflows/test-backends.yml

```diff
@@ -3,7 +3,11 @@ name: Backend Tests
 on:
   workflow_dispatch:
   push:
-    branches: [main]
+    branches: [main, backend_tests_2]
+
+concurrency:
```

Comment:
I'm not very familiar with this feature, but reading the docs indicates that the job will be canceled if any concurent part fails, but i don't think that's the behavior we want right?
For example, if AMD testing fails, we'd like the Intel testing to continue and generate a result, yes?